keepsorted/options_test.go

```diff
@@ -87,6 +87,24 @@ func TestBlockOptions(t *testing.T) {
 				PrefixOrder: []string{"a", "b", "c", "d"},
 			},
 		},
+		{
+			name:           "ItemList_YAML",
+			in:             `prefix_order=[a, b, c d, 'e",\f', 'g]', "h\"]"]`,
+			defaultOptions: blockOptions{AllowYAMLLists: true},
+
+			want: blockOptions{
+				AllowYAMLLists: true,
+				PrefixOrder:    []string{"a", "b", "c d", `e",\f`, "g]", `h"]`},
+			},
+		},
+		{
+			name: "ItemList_YAML_NotAllowed",
+			in:   `prefix_order=[a, b, c d, 'e",\f', 'g]', "h\"]"]`,
+
+			want: blockOptions{
+				PrefixOrder: []string{"[a", ""},
```

Comment:
This example perhaps should be amended into one that is actually valid today. i.e., the use case where the string is not YAML and it needs to not be parsed as YAML. Your spaces are preventing it from working right. Removing the spaces would work, though if you want to be realistic, most real world usage is stuff like "ignore_prefixes=[,[`" (i.e., ignore the square opening brace of markdown links and same for if the text of the link is code formatted).

Reply:
Done